js/WatchList.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import WatchListButton from "./AddWatchListButton";
+//
+class WatchList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      watchList: []
+    };
+  }
+  componentDidMount() {
+    let watchList = [];
+    for (let i = 0; i < localStorage.length; i++) {
```

Comment:
`localStorage.length`

![img](https://i.imgur.com/IppKJ.jpg)

How your code will work if I will add more values to `localStorage`? I mean, another developer. This construction is very fragile.